core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -513,60 +502,59 @@ private void registerBuiltinFunctionScoreParsers() {
     }
 
     private void registerBuiltinQueryParsers() {
-        registerQuery(MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent, MatchQueryBuilder.NAME,
-                "match_phrase", "matchPhrase", "match_phrase_prefix", "matchPhrasePrefix", "matchFuzzy", "match_fuzzy", "fuzzy_match");
-        registerQueryParser(MultiMatchQueryParser::new);
-        registerQueryParser(NestedQueryParser::new);
-        registerQueryParser(HasChildQueryParser::new);
-        registerQueryParser(HasParentQueryParser::new);
-        registerQueryParser(DisMaxQueryParser::new);
-        registerQueryParser(IdsQueryParser::new);
-        registerQueryParser(MatchAllQueryParser::new);
-        registerQueryParser(QueryStringQueryParser::new);
-        registerQueryParser(BoostingQueryParser::new);
+        registerQuery(MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent, MatchQueryBuilder.NAME, "match_phrase",
+                "match_phrase_prefix", "match_fuzzy", "fuzzy_match");
+        registerQuery(MultiMatchQueryBuilder.PROTOTYPE::readFrom, MultiMatchQueryBuilder::fromXContent, MultiMatchQueryBuilder.NAME);
```

Comment:
The previous builder also supported "multiMatch". I'm late to the party, but this is one problem I see in removing the alternative `names()` method from the builders, it moves the parser names (and alternatives) far away from the builders themselves. I guess this move is part of makign QueryParse a functional interface, but couldn't we leave the `names()` method and use it even though it will not be part of the QueryParser() interface? Or add it to AbstractQueryBuilder instead? Having all these string constants here in this class without a connection to their builder is a source of errors IMHO.